pokemongo_bot/cell_workers/camp_fort.py

```diff
@@ -0,0 +1,162 @@
+import math
+import time
+
+from pokemongo_bot.base_task import BaseTask
+from pokemongo_bot.cell_workers.utils import distance, coord2merc, merc2coord
+from pokemongo_bot.constants import Constants
+from pokemongo_bot.datastore import Datastore
+from pokemongo_bot.human_behaviour import random_lat_long_delta, random_alt_delta
+from pokemongo_bot.walkers.polyline_walker import PolylineWalker
+from pokemongo_bot.worker_result import WorkerResult
+
+
+class CampFort(Datastore, BaseTask):
+    SUPPORTED_TASK_API_VERSION = 1
+
+    def __init__(self, bot, config):
+        super(CampFort, self).__init__(bot, config)
+
+    def initialize(self):
+        self.destination = None
+        self.stay_until = 0
+        self.last_position_update = 0
+        self.walker = None
+
+        self.config_max_distance = self.config.get("max_distance", 1000)
+        self.config_min_forts_count = self.config.get("min_forts_count", 2)
+        self.config_min_lured_forts_count = self.config.get("min_lured_forts_count", 1)
+        self.config_camping_time = self.config.get("camping_time", 1800)
+
+    def work(self):
+        now = time.time()
+
+        if self.stay_until >= now:
+            if now - self.last_position_update >= 1:
+                lat = self.destination[0] + random_lat_long_delta() / 5
+                lon = self.destination[1] + random_lat_long_delta() / 5
+                alt = self.walker.pol_alt + random_alt_delta() / 2
+                self.bot.api.set_position(lat, lon, alt)
+                self.last_position_update = now
+                self.bot.heartbeat()
```

Comment:
I will remove it. It is not needed anymore since commit 5b3cfbb.

Otherwise, if heartbeat should be called in tick, why isn't it ?